docs/config-scheme.md

```diff
@@ -0,0 +1,377 @@
+        resValue "string", "build_config_package", "com.saigontechnolgy.rnbaseproject" // add this line
+
+        if (isNewArchitectureEnabled()) {
+            // We configure the CMake build only if you decide to opt-in for the New Architecture.
+            externalNativeBuild {
+                cmake {
+                    arguments "-DPROJECT_BUILD_DIR=$buildDir",
+                        "-DREACT_ANDROID_DIR=$rootDir/../node_modules/react-native/ReactAndroid",
+                        "-DREACT_ANDROID_BUILD_DIR=$rootDir/../node_modules/react-native/ReactAndroid/build",
+                        "-DNODE_MODULES_DIR=$rootDir/../node_modules",
+                        "-DANDROID_STL=c++_shared"
+                }
+            }
+            if (!enableSeparateBuildPerCPUArchitecture) {
+                ndk {
+                    abiFilters (*reactNativeArchitectures())
+                }
+            }
+        }
+    }
+
+   // ---
+...
+```
+
+Value `APP_ID` in env below:
+
+```bash
+API_URL=
+APP_ENV=dev
+CODEPUSH_KEY_IOS=
+CODEPUSH_KEY_ANDROID=
+APP_ID=com.saigontechnolgy.rnbaseproject.development // this line
+APP_NAME=BaseDev 
+```
+
+Names should match based on **productFlavors**, so **productiondebug** will match **debug** in this case and generate debug build of App with configuration from `.env.production`.
+
+Also add matchingFallbacks in buildTypes as shown below:
+
+`android/app/build.gradle`
+
+```bash
+buildTypes {
+        debug {
+            signingConfig signingConfigs.debug
+            matchingFallbacks = ['debug', 'release'] // <- add this line
+        }
+        release {
+            signingConfig signingConfigs.debug
+            minifyEnabled enableProguardInReleaseBuilds
+            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
+        }
+    }
+```
+
+### Android Change App name and App Icon
+
+Just copy the `android/app/main` folder and rename it to the referring names placed in the flavors in our case we put it
+**development** and **staging**.
+
+- Duplicate main file
+
+![](https://res.cloudinary.com/practicaldev/image/fetch/s--npUCbFOy--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/9mudwxn57m3puga1g698.png)
+
+- Rename file to **development** or **staging** and remove file java
+
+![](https://res.cloudinary.com/practicaldev/image/fetch/s--uj3HJFhr--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/al7sumomfwa8j6nspbuu.png)
+
+- To change the app icons, just add it inside the specific mipmap of the build development, staging or main(production).
+
+- To change app name, open file and rename
+
+`android/app/src/main/AndroidManifest.xml`
+
+```bash
+    <application
+      android:name=".MainApplication"
+      android:label="@string/APP_NAME" // change this line
+      android:icon="@mipmap/ic_launcher"
+      android:roundIcon="@mipmap/ic_launcher"
```

Comment:
Why the value for roundIcon is ic_launcher

Reply:
i've updated it